docs/source/apis/api_leveling.md

```diff
@@ -0,0 +1,57 @@
+# Llama Stack API Stability Leveling
+
+In order to provide a stable experience in Llama Stack, the various APIs need different stability levels indicating the level of support, backwards compatability, and overall production readiness.
+
+## Different Levels
+
+### v1alpha
+
+- Little to no expectation of support between versions
+- Breaking changes are permitted
+- Datatypes and parameters can break
+- Routes can be added and removed
+
+### v1beta
+
+- API routes ensured between versions
+- Parameters and return types are not
+- API, besides minor fixes and adjustments, should be _almost_ v1. Changes should not be drastic.
+
+### v1 (stable)
+
+- Considered stable
+- Backwards compatible between Z-streams
+  - Y-stream breaking changes must go through the proper approval and announcement process.
+- Datatypes for a route and its return types cannot changed between Z-streams
+  - Y-stream datatype changes should be sparing, unless the changes are additional net-new parameters
+- Must have proper conformance testing as outlined in https://github.com/llamastack/llama-stack/issues/3237
+
+
+### Approval and Announcement Process for Breaking Changes
+
+- **PR Labeling**: Any pull request that introduces a breaking API change must be clearly labeled with `breaking-change`.
```

Comment:
I'm wondering how useful this will be in practice, but I guess it doesn't hurt.

Reply:
its really just another mechanism to ensure these types of changes are purposeful and not accidental. I think applying a bit of friction here is a good thing.

Reply:
came back to this, I don't think this is useful especially if we have a proper CHANGELOG but won't block for this, will see if this happens in practice :)